launch/rover_core.launch

```diff
@@ -0,0 +1,13 @@
+<!-- 
+    This launch file runs the core nodes required for most of our other software to run.
```

Comment:
needs a better name, these are all sim-specific things and the file name should reflect that

Reply:
Are they? the rover model is definitely needed for more than just sim, it's used in rviz and surely for any arm stuff. I'm actually still a little bit unclear about the role of joint state publisher and robot state publisher on the physical rover, been meaning to research that.

Reply:
in theory the joint states should be coming from sensor info. Not super useful for the wheel joints, but e.g. the arm. The robot_state_publisher can run on the real thing to automatically get the TF hierarchy.

Reply:
Ok yeah looks like youre right, the joint state publisher should be totally replaced by ESW code on the physical rover, so I'll move that to the simulation launch file. FYI joint state is useful for wheel joints bc that's how we get encoder velocity readings.

Reply:
done